Slim/ResponseEmitter.php

```
@@ -133,7 +133,11 @@ public function isResponseEmpty(ResponseInterface $response): bool
         if (in_array($response->getStatusCode(), [204, 205, 304], true)) {
             return true;
         }
-        $contents = (string) $response->getBody();
-        return !strlen($contents);
+        $stream = $response->getBody();
+        $seekable = $stream->isSeekable();
+        if ($seekable) {
+            $stream->rewind();
+        }
+        return $seekable ? $stream->read(1) === '' : $stream->eof();
```

Could this be made easier with ... ?

```php
return $stream->getSize() > 0;
```

---

Your logic condition need to invert. `return !$stream->getSize() > 0`. Please, run unit tests with your code. You will see some errors (AppTest errors can be ignored). The most interesting is the error `Slim\Tests\ResponseEmitterTest::testRespondIndeterminateLength`. It shows that reading from a stream of at least one byte is the most reliable way. I myself don’t like such excessive complexity.

---

I found yet another example. Resource `$resource = popen('echo 12', 'r')` will be defined as empty, but in reality has content.